# Changes the name of the zlib static library from "zlibstatic" to "z" so that Ruby's build system can find it.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -201,16 +201,16 @@ if(ZLIB_BUILD_SHARED)
 endif(ZLIB_BUILD_SHARED)
 
 if(ZLIB_BUILD_STATIC)
-    add_library(zlibstatic STATIC ${ZLIB_SRCS} ${ZLIB_PUBLIC_HDRS}
+    add_library(z STATIC ${ZLIB_SRCS} ${ZLIB_PUBLIC_HDRS}
                                   ${ZLIB_PRIVATE_HDRS})
-    add_library(ZLIB::ZLIBSTATIC ALIAS zlibstatic)
+    add_library(ZLIB::ZLIBSTATIC ALIAS z)
     target_include_directories(
-        zlibstatic
+        z
         PUBLIC $<BUILD_INTERFACE:${zlib_BINARY_DIR}>
                $<BUILD_INTERFACE:${zlib_SOURCE_DIR}>
                $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
     target_compile_definitions(
-        zlibstatic
+        z
         PRIVATE ZLIB_BUILD
                 $<$<BOOL:NOT:${HAVE_FSEEKO}>:NO_FSEEKO>
                 $<$<BOOL:${HAVE___ATTR__VIS_HIDDEN}>:HAVE_HIDDEN>
@@ -218,7 +218,7 @@ if(ZLIB_BUILD_STATIC)
                 $<$<BOOL:${MSVC}>:_CRT_NONSTDC_NO_DEPRECATE>
         PUBLIC $<$<BOOL:${HAVE_OFF64_T}>:_LARGEFILE64_SOURCE=1>)
     set_target_properties(
-        zlibstatic PROPERTIES EXPORT_NAME ZLIBSTATIC OUTPUT_NAME
+        z PROPERTIES EXPORT_NAME ZLIBSTATIC OUTPUT_NAME
                                                      z${zlib_static_suffix})
 endif(ZLIB_BUILD_STATIC)
 
@@ -249,7 +249,7 @@ if(ZLIB_INSTALL)
 
     if(ZLIB_BUILD_STATIC)
         install(
-            TARGETS zlibstatic
+            TARGETS z
             COMPONENT Development
             EXPORT zlibStaticExport
             RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
